extjsdk/src/main/java/io/vantiq/extjsdk/ExtensionWebSocketListener.java

```diff
@@ -370,6 +387,19 @@ else if (msg.get("op").equals(ExtensionServiceMessage.OP_QUERY)) {
                         log.warn("Query received with no handler set");
                     }
                 }
+                else if (msg.get("op").equals(ExtensionServiceMessage.OP_RECONNECT_REQUIRED)) {
+                    if (this.reconnectHandler != null) {
+                        this.reconnectHandler.handleMessage(msg);
+                    }
+                    else if (!client.autoReconnect) {
```

Comment:
The _if_ logic here is a bit twisted.  It seems to have !autoReconnect as an alternative to having a handler.  Then the 
_else if !reconnect... if reconnect..._ seems odd.  Seems like something like

`if handler call handler

if !autoReconnect then
    whine about it
else
    do it.`

would seem more appropriate.

Reply:
Ah, OK.  Maybe just a comment in the !autoReconnect that say something about only complaining if no handler's there.  I assume that if the caller doesn't care about reconnected, they can skip the handler.  The configHandler() will still be called after the connect?

Reply:
Yep. The error message is just there for when it has no notification that it should reconnect. It will still work if the user manually calls connectToSource(), but they don't have any reason to.